.github/workflows/main.yml

```diff
@@ -0,0 +1,29 @@
+name: Docker Image CI
```

Comment:
Can we change the name of this to something like `Build and test`.  Basically, summarize the steps that are included in this workflow.